ico/tests/contracts/test_security_token.py

```diff
@@ -487,3 +504,30 @@ def test_erc865(chain, security_token, team_multisig, customer, private_key, sig
     key_raw = signed_data["r_bytes"] + signed_data["s_bytes"] + signed_data["v"].to_bytes(1, byteorder="big")
 
     security_token.transact({"from": customer}).transferPreSigned(key_raw, team_multisig, 123, 123, 123)
+
+
+def test_restricted_transfer_agent(chain, security_token, team_multisig, customer, restricted_transfer_agent, mock_kyc):
+    check_gas(chain, security_token.transact({"from": team_multisig}).setTransactionVerifier(restricted_transfer_agent.address))
+
+    check_gas(chain, security_token.transact({"from": team_multisig}).transfer(customer, 10))
+    assert security_token.call().balanceOf(customer) == 10
+
+    check_gas(chain, mock_kyc.transact({"from": team_multisig}).setFlags(customer, 0))
+
+    with pytest.raises(TransactionFailed):
+        check_gas(chain, security_token.transact({"from": team_multisig}).transfer(customer, 10))
+
+    with pytest.raises(TransactionFailed):
+        check_gas(chain, security_token.transact({"from": customer}).transfer(team_multisig, 10))
+
+    check_gas(chain, mock_kyc.transact({"from": team_multisig}).setFlags(team_multisig, 2))
+
+    check_gas(chain, security_token.transact({"from": team_multisig}).transfer(customer, 10))
+
+    check_gas(chain, mock_kyc.transact({"from": team_multisig}).setFlags(customer, 1))
+
+    with pytest.raises(TransactionFailed):
+        check_gas(chain, security_token.transact({"from": customer}).transfer(team_multisig, 10))
+
+    check_gas(chain, mock_kyc.transact({"from": team_multisig}).setFlags(team_multisig, 3))
```

Comment:
Lets have a enum for setting these flags. A programmer using the smart contract shouldn't worry about calculating what the flag would look like in binary and what its implications owuld be